Add ekobi A65 14W + Sylstar 24W

forabi opened the pull request.

bramstroker commented:
Do you have some more information about this exact light? As A65 14W seems quite generic.

Also please check the `supported_color_modes` of the light.
I see you included both `brightness` and `color_temp`, but this combination should not be possible in HA, also according to the specs here: https://developers.home-assistant.io/docs/core/entity/light/#color-modes.
You should probably only provide either `brightness` or `color_temp`. The first one when only brightness can be controlled in HA, second when it's possible to change the color temperature in HA.

forabi commented:
Thanks for the quick feedback. The light supports RGB and color temperature
adjustment. Unfortunately I could not get readings for HS mode. I'll get a
Tape P110 on Monday to see if I can get accurate values.

As for the model, it's this one: ekobi SMART LED E27 14W 1400Lm RGB CCT
WIFI TUYA, Alexa, Android, iOS, google home, Sprachsteuerung, Smartphone
App Steuerung, Mehrfarbige Birne, Dimmbar, Frequenz 2,4GHz 2700-6500K, 3-er
Pack https://amzn.eu/d/8r246V6

Bram Gerritsen ***@***.***> schrieb am So., 3. Sept. 2023,
08:41:

> Do you have some more information about this exact light? As A65 14W seems
> quite generic.
>
> Also please check the supported_color_modes of the light.
> I see you included both brightness and color_temp, but this combination
> should not be possible in HA, also according to the specs here:
> https://developers.home-assistant.io/docs/core/entity/light/#color-modes.
> You should probably only provide either brightness or color_temp. The
> first one when only brightness can be controlled in HA, second when it's
> possible to change the color temperature in HA.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bramstroker/homeassistant-powercalc/pull/1852#issuecomment-1704027100>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAHXIGOUVXY4BZPW74MPPELXYQRBVANCNFSM6AAAAAA4JBWBPY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

bramstroker commented:
In that case you'll need to provide both `hs` and `color_temp`.
Let's await till you have the Tapo, hopefully that one will be able to get some measurements.

You can set `SMART LED E27 14W 1400Lm RGB CCT` in name property of model.json as that will be shown in supported model listing.

forabi commented:
@bramstroker I've updated this to use Tapo P110 and used color_temp and hs modes. I've also added a new ceiling lamp which is [this model](https://www.amazon.de/-/en/gp/product/B0B2DR53KR/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1)

bramstroker commented:
@forabi Thanks.
In the future, could you please create a single PR per light? You'll need to make a new branch in git for that. plenty of tutorials to found on internet for that.

Your aliases seems off. These are only needed for auto discovery. I see you put a very long name in it, that can't be correct.
Please have a look into device page of the light, and the information on the top left.

<img width="344" alt="Screenshot 2023-09-09 at 09 12 16" src="https://github.com/bramstroker/homeassistant-powercalc/assets/2345875/c7e82508-4938-4bc8-b149-327cf1a55615">

Could you share this information for both of the lights please?

forabi commented:
@bramstroker Updated the aliases

bramstroker commented:
Thanks. Still left with some questions though.
As I cannot find anything about `uebyc181yivkcndi` and `bcyytn6qryldlgax` on Google, seems not an actual model or something we can use to autodiscover the same model on other users sytems.
`Smart Ceiling Lamp` seem also way to generic and not an actual model identifier.

This whole Tuya ecosystem is such a big mess unfortunately.

Could you share the device information as seen in HA as shown on my screenshot in previous comment?
Also which integration are you using to get these lights into Home Assistant?

Moayyad-Zakkar commented:
@bramstroker This is a Tuya model ID, the auto discovery works with my branch just fine.

forabi commented:
@bramstroker This is a Tuya model ID, the auto discovery works with my branch just fine.

<img width="1108" alt="image" src="https://github.com/bramstroker/homeassistant-powercalc/assets/1012761/9a36c219-aa08-458d-83f4-dbf46ccb1d38">
<img width="1108" alt="image" src="https://github.com/bramstroker/homeassistant-powercalc/assets/1012761/e2f6e032-c794-4528-a837-d2941550b923">

bramstroker commented:
For the Sylstar light according to following amazon link the model is `‎SL-CL-24`, which seems an actual model number handled by the brand. So I suggest to change the directory name from `Smart Ceiling Lamp` to `SL-CL-24` so we have correct model in supported model listing: https://github.com/bramstroker/homeassistant-powercalc/blob/master/docs/supported_models.md

https://www.amazon.co.uk/SYLSTAR-Ceiling-Dimmable-Assistant-Bathroom/dp/B09HH1GTQK

<img width="873" alt="Screenshot 2023-09-09 at 19 42 28" src="https://github.com/bramstroker/homeassistant-powercalc/assets/2345875/50928b41-7c2b-4c43-bb84-133a80847d18">

bramstroker commented:
> @bramstroker This is a Tuya model ID, the auto discovery works with my branch just fine.
> 
> <img alt="image" width="1108" src="https://user-images.githubusercontent.com/1012761/266795228-9a36c219-aa08-458d-83f4-dbf46ccb1d38.png"> <img alt="image" width="1108" src="https://user-images.githubusercontent.com/1012761/266795245-e2f6e032-c794-4528-a837-d2941550b923.png">

I don't know Tuya that well as I don't have any devices myself, but maybe this hash is unique per Tuya device in each unique Tuya installation?
It's enough to only add `uebyc181yivkcndi` and `bcyytn6qryldlgax` as alias because Powercalc will check value between parentheses as fallback.
But when this hash is unique per installation it will work for your HA installation, but not for others, so makes no sense to add the alias in that case. But not sure how to know this for sure.

bramstroker commented on custom_components/powercalc/data/tuya/A65 14W/model.json:
```diff
@@ -0,0 +1,14 @@
+{
+  "calculation_strategy": "lut",
+  "measure_description": "Measured with utils/measure script",
+  "measure_device": "P110",
```
```suggestion
  "measure_device": "Tapo P110",
```

bramstroker commented on custom_components/powercalc/data/tuya/A65 14W/model.json:
```diff
@@ -0,0 +1,14 @@
+  },
+  "name": "A65 14W \u4e94\u8def BK-N \u516c\u7248",
+  "standby_power": 0.46,
+  "aliases": ["uebyc181yivkcndi", "A65 14W 五路 BK-N 公版"]
```
```suggestion
  "aliases": ["uebyc181yivkcndi"]
```

bramstroker commented on custom_components/powercalc/data/tuya/Smart Ceiling Lamp/model.json:
```diff
@@ -0,0 +1,14 @@
+{
+  "calculation_strategy": "lut",
+  "measure_description": "Measured with utils/measure script",
+  "measure_device": "P110",
```
```suggestion
  "measure_device": "Tapo P110",
```

bramstroker commented on custom_components/powercalc/data/tuya/Smart Ceiling Lamp/model.json:
```diff
@@ -0,0 +1,14 @@
+  },
+  "name": "SYLSTAR LED Ceiling Light 24W",
+  "standby_power": 0.73,
+  "aliases": ["bcyytn6qryldlgax", "Smart Ceiling Lamp"]
```
```suggestion
  "aliases": ["bcyytn6qryldlgax"]
```

forabi commented:
@bramstroker I believe you are right, this is probably the device virtual ID which is unique per each device "boostrap" process. I'll see if I can find some more stable model ID

forabi commented:
@bramstroker Found this in the diagnostics file for Tuya integration:

```
    "name": "Bedroom Lamp",
    "model": "SYS-C24W-CW",
    "category": "dj",
    "product_id": "bcyytn6qryldlgax",
    "product_name": "Smart Ceiling Lamp",
```

forabi commented:
The other lamp:

```
    "name": "Bulb 9",
    "model": "",
    "category": "dj",
    "product_id": "uebyc181yivkcndi",
    "product_name": "A65  14W    \u4e94\u8def   BK-N   \u516c\u7248 ",
```

bramstroker commented:
> @bramstroker I believe you are right, this is probably the device virtual ID which is unique per each device "boostrap" process. I'll see if I can find some more stable model ID

Ah, that clarifies why I couldn't find any reference on Google.
But when this is the only information which is exposed to HA we cannot use it for auto discovery.
So I don't think any stable model id is exposed to HA somehow which can be utilized.
That being said we can still adopt the lights into the DB, but the user must set it up themself using this procedure: https://homeassistant-powercalc.readthedocs.io/en/latest/sensor-types/virtual-power-library.html.

bramstroker commented:
Not really much more to work with unfortunately looking at those diagnostics.
That very unfortunate but we have to work with what is available.
We cannot provide reliable autodiscovery for these lights. I might add a boolean to model.json in the future to indicate that, so we can also list in the supported model list that these lights cannot be autodiscovered.

To get the PR wrapped up following needs to be resolved:

- You may choose `SYS-C24W-CW` or `SL-CL-24` for the directory name of the Sylstar
- aliases must be removed from the model.json as these are not applicable to other users of powercalc
- Other suggestion `P110` -> `Tapo P110` applied.
- Name of the ekobi light set to `ekobi SMART LED E27 14W 1400Lm RGB CCT`

I was still search for an actual model id for the ekobi, but was not able to find any.
Maybe on the package? But I doubt you still have that.

forabi commented:
Ekobi
![SWT20230909_201107_SWTLmcr17dreamshot.jpg](https://github.com/bramstroker/homeassistant-powercalc/assets/1012761/67609b02-4b60-45b8-be3d-9e8163ecd8b5)

forabi commented:
@bramstroker Addressed all feedback, also renamed ekobi's folder to the model name on the package

bramstroker commented:
Thanks for your efforts. Looks good now.
